src/sysl/exporters/json_out/serializer.py

```diff
@@ -88,75 +88,155 @@ def serializer(context):
         for (tname, t) in sorted(app.types.iteritems()):
             if not t.WhichOneof('type') in ['relation', 'tuple']:
                 continue
-
-            w()
-            with java.Method(w, 'public', 'void', 'serialize' + tname + 'View',
-                             [('JsonGenerator', 'g'),
-                              (tname + '.View', 'view')],
-                             throws=['IOException']):
-                with java.If(w, 'view.isEmpty()'):
-                    w('return;')
-                w('g.writeArrayFieldStart("{}");', tname)
-                with java.For(w, '{} item : view', tname):
-                    w('g.writeStartObject();')
+            java.SeparatorComment(w)
```

Comment:
This is a pretty big and complex change and very tricky to understand for me (even before these changes this was challenging code).
Could you please at least break up the `serializer` and `deserializer` functions into smaller junks if possible so that they become less than 200 line functions?
For details of this change I'll have to rely on @anzdaddy

Reply:
Have you had a look @anzdaddy?

Reply:
No, and I probably won't have a chance to do so till Friday at the earliest.